# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.cpp
# Commit: 941ad27caa8e
# Full Hash: 941ad27caa8e1c3782106e2b64e5315a0d282342
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-12-07 09:10:49
# Description:
#   Bug 1935475 - Defend against sync callbacks in Show call. r=edenchuang
#   
#   The crash in this bug seems to indicate that we are receiving a sync
#   observer notification callback during our call to Show() and this is
#   resulting in Show() returning an error code.  Because the callbacks
# ==============================================================================

diff -r bbcb761190c0 -r 941ad27caa8e dom/notification/NotificationParent.cpp
--- a/dom/notification/NotificationParent.cpp	Sat Dec 07 01:24:02 2024 +0200
+++ b/dom/notification/NotificationParent.cpp	Fri Dec 06 19:28:31 2024 +0000
@@ -152,7 +152,9 @@
   // nsIAlertNotification::LoadImage)
   // Step 4.3: Run the show steps for notification.
   nsresult rv = Show();
-  if (NS_FAILED(rv)) {
+  // It's possible that we synchronously received a notification while in Show,
+  // so mResolver may now be empty.
+  if (NS_FAILED(rv) && mResolver) {
     mResolver.take().value()(CopyableErrorResult(rv));
   }
   // If not failed, the resolver will be called asynchronously by
